pkg/virt-handler/vm.go

```diff
@@ -16,10 +17,21 @@ func NewVMController(listWatcher cache.ListerWatcher, domainManager libvirt.Doma
 			fmt.Printf("VM ADD\n")
 			vm := obj.(*v1.VM)
 			err := domainManager.SyncVM(vm)
+
 			if err != nil {
 				fmt.Println(err)
 				recorder.Event(vm, kubev1.EventTypeWarning, v1.SyncFailed.String(), err.Error())
 				return cache.ErrRequeue{Err: err}
+			} else {
+				restClient, err := kubecli.GetRESTClient()
+				if err != nil {
+					return cache.ErrRequeue{Err: err}
+				}
+				vm.Status.Phase = v1.Succeeded
+				err = restClient.Put().Resource("vms").Body(vm).Name(vm.ObjectMeta.Name).Namespace(kubeapi.NamespaceDefault).Do().Error()
```

Comment:
For kubernetes internal things there would be PATCH, but not for TPR and we have to make sure to only update if needed anyway. There are not that much updates if we do it right.